File: examples/java/product/src/main/java/dev/angzarr/examples/product/ProductState.java
package dev.angzarr.examples.product;

public record ProductState(
    String sku,
    String name,
    String description,
    int priceCents,
    String status
) {
    public static ProductState empty() {
        return new ProductState("", "", "", 0, "");
    }

    public boolean exists() {
        return !sku.isEmpty();
    }

    public boolean isActive() {
        return "active".equals(status);
    }

    public boolean isDiscontinued() {
        return "discontinued".equals(status);
    }
}

File: examples/kotlin/product/src/main/kotlin/dev/angzarr/examples/product/ProductState.kt
package dev.angzarr.examples.product

data class ProductState(
    val sku: String = "",
    val name: String = "",
    val description: String = "",
    val priceCents: Int = 0,
    val status: String = ""
) {
    fun exists(): Boolean = sku.isNotEmpty()
    fun isActive(): Boolean = status == "active"
    fun isDiscontinued(): Boolean = status == "discontinued"

    companion object {
        fun empty(): ProductState = ProductState()
    }
}
